File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/java/io/github/lishangbu/avalon/dataset/entity/BerryFirmness.java
package io.github.lishangbu.avalon.dataset.entity;

import io.github.lishangbu.avalon.hibernate.Flex;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serial;
import java.io.Serializable;
import lombok.Data;

/// 树果硬度(BerryFirmness)实体类
///
/// 表示树果的硬度分类，用于描述树果的质地
///
/// @author lishangbu
/// @since 2025/08/20
@Data
@Entity
@Table(comment = "树果硬度")
public class BerryFirmness implements Serializable {
    @Serial private static final long serialVersionUID = 1L;

    /// 主键
    @Id
    @Flex
    @Column(comment = "主键")
    private Long id;

    /// 内部名称
    @Column(comment = "内部名称", length = 100)
    private String internalName;

    /// 树果硬度名称
    @Column(comment = "树果硬度名称", length = 100)
    private String name;
}

File: avalon-modules/avalon-dataset/avalon-dataset-model/src/main/kotlin/io/github/lishangbu/avalon/dataset/entity/BerryFirmness.kt
package io.github.lishangbu.avalon.dataset.entity

import io.github.lishangbu.avalon.jimmer.id.SnowflakeIdGenerator
import org.babyfish.jimmer.jackson.JsonConverter
import org.babyfish.jimmer.jackson.LongToStringConverter
import org.babyfish.jimmer.sql.Column
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.Table

@Entity
@Table(name = "berry_firmness")
interface BerryFirmness {
    @Id
    @GeneratedValue(generatorType = SnowflakeIdGenerator::class)
    @JsonConverter(LongToStringConverter::class)
    val id: Long

    @Column(name = "internal_name")
    val internalName: String?

    @Column(name = "name")
    val name: String?
}
